DO NOT ACCEPT

gm-infinite opened the pull request:
-branch gözüksün diye bu merge requesti atıyorum, testing için hazır ama mergelemek için değil.

gm-infinite committed:
started on  || && parser

- first created a way to see the recursion tree thats created, the tree's leaf nodes are th einputs ıll send to the next parser in order from left to right while respecting && priority.
-fixed makefile to actually compile the project
-tested everytihng using the main input system, so there werent any unit testing for this addition.

gm-infinite committed:
enhanced the parser

-added paranthesis support, so when && or || comes with paranthesis priority, itll be taken to account first
-made my code less complicated to follow and will continue to do so.
#found a pronlem in split, the spliting also happenes when spance characters are in between " ",which it shouldnt be.

gm-infinite committed:
for got the complitations in the file

deleted minishell and libft.a files

gm-infinite committed:
seem like its nearly finished

-added a file to store fucntions tthat are used outside of and_or_parser (also for 5 function limit), or the functiond that I think might be used later on.
-removed paranthesis parity from main since it is not required for this project (needed ";" handling, which we are told to not implement)
-finally paranthesis priority works, ım sending this with testing still inside since we might need it later and I still need to test alot to confirm it works with all cases.
-did NOt make it norminette yet, this is a test build, when we confirm it works ıll make it norm, remove testing functions and push it and merge it.

thecloudrazor reviewed (commented):
# Vakti gelince mergelenebilir. ✅️
Çalışmalara devam ...

gm-infinite committed:
99% done

-found a segmentation fault, and added a defult initilizer for shell struct to fix it.
-other than stuff I would consider "extra", theres nothing more to do it seems. maybe some comments can be changed to fit better to the norm.
- the code is sdited to fit norm rules and it passses the norminette tests (other than intantional comments we left)

Commit:
-changed is_empty.c to main_utils.c
-gave an aditional note to tasks, it seems like we misunderstood something

Commit:
-added extra notes section at the ned of tasks.md file
-this section is for extra dicoveries that we made while doing this project

Commit:
tasks mistake
-there was a mistake while adding extra notes section

thecloudrazor committed:
Improve comments in parser files for clarity.  Modify TASKS.md

thecloudrazor committed:
Update TASKS.md: parser implementation for `||` and `&&` as complete, add additional notes, improve formatting

thecloudrazor committed:
Improve comment formatting in main_utils.c for safe_exit function
